.github/workflows/deploy.yml

```diff
@@ -59,11 +60,12 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python 3.7
-      uses: actions/setup-python@v2
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
       with:
-        python-version: 3.9
+        python-version: 3.10
+        cache: pip
```

Comment:
what's the reason behind caching pip?

Reply:
Because I think that should speed things up a little. There is a catch though, I'm not 100% sure we'll ever get the newest versions of our dependencies this way, since we don't list explicit version numbers in our requirements. And those requirements files are used to create a hash-index for the cache. Maybe rip it out again after all?